modules/nf-core/comebin/runcomebin/main.nf

```diff
@@ -0,0 +1,75 @@
+process COMEBIN_RUNCOMEBIN {
+    tag "${meta.id}"
+    label 'process_high'
+
+    conda "${moduleDir}/environment.yml"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/comebin:1.0.4--hdfd78af_0':
+        'biocontainers/comebin:1.0.4--hdfd78af_0' }"
+
+    input:
+    tuple val(meta), path(assembly), path(bam, stageAs: "bam/*")
+
+    output:
+    tuple val(meta), path("${prefix}/comebin_res_bins/*.fa.gz"), emit: bins
+    tuple val(meta), path("${prefix}/comebin_res.tsv")         , emit: tsv
+    tuple val(meta), path("${prefix}/comebin.log")             , emit: log
+    tuple val(meta), path("${prefix}/embeddings.tsv")          , emit: embeddings
+    tuple val(meta), path("${prefix}/covembeddings.tsv")       , emit: covembeddings
+    path "versions.yml"                                        , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args               = task.ext.args ?: ''
+    prefix                 = task.ext.prefix ?: "${meta.id}"
+    def clean_assembly     = assembly.toString() - ~/\.gz$/
+    def decompress_contigs = assembly.toString() == clean_assembly ? "" : "gunzip -q -f $assembly"
+    def cleanup            = decompress_contigs ? "rm ${clean_assembly}" : ""
+    """
+    ${decompress_contigs}
+
+    cp ${clean_assembly} . # ISSUE: temporary files will be generated in the directory of the assembly file, following links, copying prevents that
+
+    run_comebin.sh \\
+        -t ${task.cpus} \\
+        -a ${clean_assembly} \\
+        -p bam/ \\
+        -o . \\
+        $args
+
+    mv comebin_res ${prefix}
+
+    find ${prefix}/comebin_res_bins/*.fa -exec gzip {} \\;
+
+    for filename in ${prefix}/comebin_res_bins/*.fa.gz; do mv "\${filename}" "${prefix}/comebin_res_bins/${prefix}.\$(basename \${filename})"; done; # avoid file name collisions
```

Comment:
An issue occurred without that line in the downstream module saying:
```
Process `NFCORE_MAG:MAG:BINNING:SEQKIT_STATS` input file name collision -- There are multiple input files for each of the following file names: 0.fa.gz
```
and indeed the bins were in files with name `<int>.fa.gz` in folder `${prefix}/comebin_res_bins`. This line renames each file so that its `${prefix}.<int>.fa.gz` which prevents any file name collisions. 
Do you have here any more elegant solutions?